### DISCUSSION TITLE
Log KeyDown events together with video timestamps

### DISCUSSION THREAD

**Discussion by horsto** [bonsai-rx/bonsai#953] [Q&A] [answered]
Apologies if this has been answered before (I saw related discussions on the google group). I am recording video frames from a camera and create timestamps for each recorded frame. I want to export a .csv file where the first column is index of each frame, the second is timestamp of each frame, and the third is zero or NONE when no key was pressed, but shows the key when it was pressed during the acquisition of the current frame. In the attached screenshot I am interested in whether either one of two buttons was pressed during an acquired frame. I am stuck here since no matter what I do / which logic I implement, the CSV file just contains data when a button was pressed, but does not keep a running log of all frames like I want. 
<img width="358" alt="workflow_button" src="https://user-images.githubusercontent.com/7548928/176791714-856780f9-261d-4084-a668-3555ec183000.PNG">

**Comment by bruno-f-cruz** [comment, marked as answer]
Heya
I would not advise you use this strategy. You will end up saving an uncessary amount of data. I would probably just save the current frame on each keypress and, post-hoc , align everything. 

Nevertheless, if you really want to do it the way you described, check the attached workflow.
[keylogging.zip](https://github.com/bonsai-rx/bonsai/files/9028052/keylogging.zip)

Hope it helps!

**Reply by horsto** [reply]
Thanks a lot @bruno-f-cruz ! It *almost* works: It just doesn't start recording anything into the .csv until I actually press a button once. Is there a way to make it start logging from the very start? 
Attaching my workflow here. 
[keylogging.zip](https://github.com/bonsai-rx/bonsai/files/9031381/keylogging.zip)

**Reply by bruno-f-cruz** [reply]
That s because you removed the string operator that sets the initial state. Add it back and it will work. :)

**Reply by horsto** [reply]
ooops! Thanks!
